lib/yargs.ts

```diff
@@ -351,7 +355,7 @@ export function Yargs (processArgs: string | string[] = [], cwd = process.cwd(),
     K extends keyof Options[T] & string = keyof Options[T] & string,
     V extends ValueOf<Options[T]> = ValueOf<Options[T]>
   > (
-    builder: (key: K, value: V, ...otherArgs: any[]) => YargsInstance,
+    builder: (key: K, value: V, ...otherArgs: any[]) => YargsInstance<T>,
```

Comment:
The wrong T is used here.

R (for result) should be used in this whole genericity introduction instead of T which is already used in several places

Reply:
Fixed